src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -19,20 +19,20 @@ public class SoftwareSpi : SpiDevice
         private readonly int _mosi;
         private readonly int _cs;
         private readonly SpiConnectionSettings _settings;
+        private readonly bool _shouldDispose;
         private GpioController _controller;
-        private bool _shouldDispose;
 
         /// <summary>
         /// Software implementation of the SPI.
         /// </summary>
         /// <param name="clk">Clock pin.</param>
-        /// <param name="miso">Master Input Slave Output pin.</param>
+        /// <param name="miso">Master Input Slave Output pin. Optional, set to -1 to ignore</param>
         /// <param name="mosi">Master Output Slave Input pin.</param>
-        /// <param name="cs">Chip select pin (or negated chip select).</param>
+        /// <param name="cs">Chip select pin (or negated chip select). Optional, set to -1 to ignore. Note that there is a ChipSelectLine in the SPIConnectionSettings as well, either will be used.</param>
         /// <param name="settings">Settings of the SPI connection.</param>
         /// <param name="controller">GPIO controller used for pins.</param>
         /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
-        public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings settings = null, GpioController controller = null, bool shouldDispose = true)
+        public SoftwareSpi(int clk, int miso, int mosi, int cs = -1, SpiConnectionSettings settings = null, GpioController controller = null, bool shouldDispose = true)
```

Comment:
bummer that miso is before mosi, we cannot change this right now especially they have the same type

Reply:
Yea I know, I was thinking of adding an overloaded constructor, but figured it would be more confusing as the order would change with the same parameter types.